app/helpers/languages_helper.rb

```diff
@@ -1,7 +1,10 @@
 # frozen_string_literal: true
 
 module LanguagesHelper
-  ISO_639_1 = {
+  # These locales can be selected as posting languages and filtered for by users.
+  # When adding a locale, prefer using a ISO-639-1 (2-letter) locale over
+  # using a ISO-639-3 (3-letter) locale if available for the language.
+  SUPPORTED_LOCALES = {
```

Comment:
This seems like a much better approach than maintaining our own version of ISO lists. Alternative would be to find a gem or other maintained data source that actually has the correct ISO datasets and import them, instead of our own attempt to mirror those.

I also like the comments throughout -- every time I look at this area I need to re-reason my through which things are used for i18n, which for translation/posting/filtering, etc. I wonder about being even more explicit in that regard in this file to show how/where each constants values are used.

Reply:
The gem to use would be https://github.com/ruby-i18n/ruby-cldr but it's not production ready.

Reply:
I have been thinking some more about this - some languages are so marginalized that they don't have their native name in the CLDR, so pulling from the CLDR exclusively would not do the trick. So, using the CLDR will probably make the code more complicated.

Also, we can't operate with just 1 list because @Gargron wants to control what goes into the posting language filters, e.g. treat American and British English as the same language there to avoid overfiltering. We also can't simply determine this by whether the locale has a country code in it, because most languages originating in China are `zh`, no matter whether they are actually Chinese or not.

What we could do is write a utility script to pull the available locales from the CLDR and merge those in. Such a script would then be run twice a year when the a CLDR version comes out. I could whip something up in Python if this is desired.